ikarus/finiteelements/mechanics/loads/traction.hh

```diff
@@ -0,0 +1,192 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * @file loads.hh
+ * @brief Definition of the Loads class for application of loads for finite element analysis.
+ * @ingroup mechanics
+ */
+
+#pragma once
+
+#  include <dune/fufem/boundarypatch.hh>
+#  include <dune/localfefunctions/impl/standardLocalFunction.hh>
+
+#  include <ikarus/finiteelements/febases/powerbasisfe.hh>
+#  include <ikarus/finiteelements/fehelper.hh>
+#  include <ikarus/finiteelements/ferequirements.hh>
+#  include <ikarus/utils/defaultfunctions.hh>
+
+namespace Ikarus {
+
+    /**
+     * @brief Loads class represents different types of loads that can be applied.
+     *
+     * @ingroup mechanics
+     *
+     * @tparam DisplacementBasedElement The type of the displacement-based finite element.
+     */
+    template<typename Basis_, typename FERequirements_ = FERequirements<>, bool useEigenRef = false>
+    class Traction : public PowerBasisFE<typename Basis_::FlatBasis> {
```

Comment:
To resolve this the best way and not introduce circlular dependencies. and to also allow a call to a given displacement function. I think we still should go with:

```cpp
template<typename DisplacementBasedFE>
    class Traction : ...
```

Reply:
and then you can call the functions you need. and `globalFlatIndices` works out of the box, even when there would be also solutions to the current situation.